File: frontend/src/api/message.ts
import http from './http'

import type { ApiResponse } from '@/types/api'

// ===== 类型定义 =====

export type MessageType = 1 | 2 | 3

export interface MessageUserInfo {
  id: string
  username: string
  nickname: string
  avatar: string
}

export interface LastMessageInfo {
  id: string
  messageType: MessageType
  content: string
  mediaUrl: string
  createdAt: string
}

export interface ConversationInfo {
  targetUser: MessageUserInfo
  lastMessage: LastMessageInfo
  unreadCount: number
  updatedAt: string
}

export interface ConversationListResp {
  list: ConversationInfo[]
  total: number
  page: number
  pageSize: number
}

export interface MessageInfo {
  id: string
  fromUserId: string
  toUserId: string
  messageType: MessageType // 1=文字 2=图片 3=视频
  content: string
  mediaUrl: string
  isRead: boolean
  createdAt: string
}

export interface MessageListResp {
  messages: MessageInfo[]
  nextCursorToken: string
  hasMore: boolean
}

interface SendMessagePayloadBase {
  toUserId: string
}

export interface SendTextMessagePayload extends SendMessagePayloadBase {
  messageType: 1
  content: string
  mediaUrl?: never
}

export interface SendMediaMessagePayload extends SendMessagePayloadBase {
  messageType: 2 | 3
  mediaUrl: string
  content?: string
}

export type SendMessagePayload = SendTextMessagePayload | SendMediaMessagePayload

// ===== 接口调用 =====

/** 获取当前用户的会话列表（offset 分页） */
export const getConversations = (params: { page?: number; pageSize?: number } = {}) =>
  http.get<ApiResponse<ConversationListResp>>('/conversations', { params })

/** 获取与指定用户的消息列表（游标分页） */
export const getMessages = (userId: string, params: { cursor?: string; count?: number } = {}) =>
  http.get<ApiResponse<MessageListResp>>(`/conversations/${userId}/messages`, { params })

/** 标记与指定用户的会话为已读 */
export const markConversationRead = (userId: string) =>
  http.put<ApiResponse<{ readCount: number }>>(`/conversations/${userId}/read`)

/** 发送私信 */
export const sendMessage = (payload: SendMessagePayload) =>
  http.post<ApiResponse<MessageInfo>>('/messages', payload)

/** 删除私信 */
export const deleteMessage = (id: string) =>
  http.delete<ApiResponse<Record<string, never>>>(`/messages/${id}`)

/** 获取未读消息总数 */
export const getUnreadCount = () =>
  http.get<ApiResponse<{ unreadCount: number }>>('/messages/unread-count')

File: flashvid-platform-gin/internal/model/message.go
package model

// MessageUserInfo 私信中对方用户简要信息
type MessageUserInfo struct {
	ID       int64  `json:"id"`
	Username string `json:"username"`
	Nickname string `json:"nickname"`
	Avatar   string `json:"avatar"`
}

// LastMessageInfo 会话中最后一条消息的摘要
type LastMessageInfo struct {
	ID          int64  `json:"id"`
	MessageType int32  `json:"messageType"`
	Content     string `json:"content"`
	MediaURL    string `json:"mediaUrl"`
	CreatedAt   string `json:"createdAt"`
}

// ConversationInfo 会话列表条目
type ConversationInfo struct {
	TargetUser  MessageUserInfo `json:"targetUser"`
	LastMessage LastMessageInfo `json:"lastMessage"`
	UnreadCount int32           `json:"unreadCount"`
	UpdatedAt   string          `json:"updatedAt"`
}

// ConversationListOutput 会话列表分页输出
type ConversationListOutput struct {
	List     []ConversationInfo `json:"list"`
	Total    int64              `json:"total"`
	Page     int                `json:"page"`
	PageSize int                `json:"pageSize"`
}

// MessageInfo 单条私信
type MessageInfo struct {
	ID          int64  `json:"id"`
	FromUserID  int64  `json:"fromUserId"`
	ToUserID    int64  `json:"toUserId"`
	MessageType int32  `json:"messageType"`
	Content     string `json:"content"`
	MediaURL    string `json:"mediaUrl"`
	IsRead      bool   `json:"isRead"`
	CreatedAt   string `json:"createdAt"`
}

// MessageListOutput 消息列表游标分页输出
type MessageListOutput struct {
	Messages        []MessageInfo `json:"messages"`
	NextCursorToken string        `json:"nextCursorToken"`
	HasMore         bool          `json:"hasMore"`
}

// SendMessageInput 发送私信的业务入参
type SendMessageInput struct {
	ToUserID    int64
	MessageType int32
	Content     string
	MediaURL    string
}
